management-app/src/api/usersAPI.ts

```diff
@@ -0,0 +1,115 @@
+/* eslint-disable no-console */
+import { BaseAPI } from './baseAPI';
+import { USERS_API_ERRORS } from './errors';
+import { TUser, TUserBase } from './types';
+
+class UsersAPI extends BaseAPI {
+  updateUser(
+    userId: string,
+    token: string,
+    dataUser: TUser,
+    updateUserCb: (data: TUser) => void,
+    errorCb = (_error: Error) => {
+      console.error(_error);
+    }
+  ) {
+    this.put(`users/${userId}`, dataUser, {
+      Authorization: `Bearer ${token}`,
+    })
+      .then((result) => {
+        BaseAPI.handleError(result, USERS_API_ERRORS);
+        return result.json();
+      })
+      .then((result) => updateUserCb(result))
+      .catch((error) => errorCb(error));
+  }
```

Comment:
If we use something like this we'll be able to use promise based approach instead of a legacy callback based one:

```suggestion
  updateUser({ userId, token, user }: { userId: string; token: string; user: TUser; }) {
    return this.put(`users/${userId}`, user, {
      Authorization: `Bearer ${token}`,
    })
      .then((result) => {
        BaseAPI.handleError(result, USERS_API_ERRORS);
        return result.json();
      });
  }
```